python/tvm/relay/op/_transform.py

```diff
@@ -204,3 +204,68 @@ def take_shape_func(attrs, inputs, out_ndims):
             axis += data_ndim
         assert 0 <= axis < data_ndim
         return [_take_with_axis_shape_func(*inputs, convert(axis), out_ndims[0])]
+
+@script
+def _argwhere_shape_func_2d(condition):
```

Comment:
I think we can flatten the condition shape, so that we don't need separate routings for different condition shapes. For example, if condition shape is (N, M, K), we can create a flatten array (N * M * K, M * K, K). Then iterate through N * M * K. Each index of condition can be computed with flatten array.

Reply:
I'm not sure whether hybrid script supports condition[index_array]. If not, we might need to use ir_builder.

Reply:
I discussed with @icemelon9 , it's not convenient to use `ir_builder` since the inputs are `Tensor`, while `ir_builder` expects `Buffer`.

Reply:
Yes. To use ir_builder we need to create buffer from tensor and use extern call, like https://github.com/dmlc/tvm/blob/3d74b48f458d23c7e95a88b82f0230348a33e767/topi/python/topi/vision/nms.py#L169 For now I think it is fine we keep enumerate style shape func, since in topi we are using the same patten as well. Later we can consider how to make both topi compute and shape func more general.

Reply:
@kevinthesun sounds good. thank you!